src/test/java/seedu/address/logic/commands/internships/role/EditInternshipRoleCommandTest.java

```diff
@@ -0,0 +1,265 @@
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.CommandAssignmentTestUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.commands.internship.role.EditInternshipRoleCommand;
+import seedu.address.logic.commands.internship.role.FindInternshipRoleCommand;
+import seedu.address.logic.commands.internship.task.FindInternshipTaskCommand;
+import seedu.address.model.AddressBook;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.fields.ApplicationOutcome;
+import seedu.address.model.fields.Cycle;
+import seedu.address.model.fields.Description;
+import seedu.address.model.fields.Location;
+import seedu.address.model.fields.Outcome;
+import seedu.address.model.fields.Pay;
+import seedu.address.model.internship.role.InternshipRole;
+import seedu.address.model.internship.role.InternshipRoleNameContainsKeywordsPredicate;
+import seedu.address.model.internship.task.InternshipTask;
+import seedu.address.model.internship.task.InternshipTaskNameContainsKeywordsPredicate;
+import seedu.address.testutil.InternshipRoleBuilder;
+import seedu.address.testutil.InternshipTaskBuilder;
+
+public class EditInternshipRoleCommandTest {
+    private final InternshipRole role1 = new InternshipRoleBuilder().withName("Company1")
+            .withRole("SWE").withCycle("Summer").build();
+
+    private final InternshipRole role2 = new InternshipRoleBuilder().withName("Company2")
+            .withRole("SWE").withCycle("Winter").build();
+
+    private final InternshipTask task1BelongingToRole1 = new InternshipTaskBuilder()
+            .withInternshipRole(role1).withTaskName("Interview Preparation_1").build();
+
+    private final InternshipTask task2BelongingToRole1 = new InternshipTaskBuilder()
+            .withInternshipRole(role1).withTaskName("OA_1").build();
+
+    private final InternshipTask task1BelongingToRole2 = new InternshipTaskBuilder()
+            .withInternshipRole(role2).withTaskName("Interview Preparation_2").build();
+
+    private final InternshipTask task2BelongingToRole2 = new InternshipTaskBuilder()
+            .withInternshipRole(role2).withTaskName("OA_2").build();
+
+    private ModelManager model;
+
+    private AddressBook getAddressBook() {
+        AddressBook ab = new AddressBook();
+        ab.addInternshipRole(role1);
+        ab.addInternshipRole(role2);
+        ab.addInternshipTask(task1BelongingToRole1);
+        ab.addInternshipTask(task2BelongingToRole1);
+        ab.addInternshipTask(task1BelongingToRole2);
+        ab.addInternshipTask(task2BelongingToRole2);
+        return ab;
+    }
+
+    private void editRolesAndTask(InternshipRole roleToBeEdited, InternshipRole editedRole, ModelManager model) {
+        for (InternshipRole r : model.getFilteredInternshipRoleList()) {
+            if (r.equals(roleToBeEdited)) {
+                model.setInternshipRole(r, editedRole);
+            }
+        }
```

Comment:
Is it possible to just use `setInternshipRole(roleToBeEdited, editedRole)`? Or does `setInternshipRole` not recognise the `roleToBeEdited` because they're not the same object?

Either way I think if you don't want to change this part is fine.